### PR TITLE
Fix request error handling

### LINKED ISSUE #208: APIs requests don't error out
If an API request returns an error (unable to fulfill order, rate limited, etc), the client will silently return the error payload without erroring out, which makes for an odd exception handling. Is this by design? Sorry if this issue is a duplicate

### PR DESCRIPTION
**Pull request by rmm5t**
## What does it do?

- Fixes error handling expectations as documented in #208
- Non-2xx HTTP status codes will now (correctly) return an error

## Related Issues

- Fixes #208
- Closes #209 (this PR is stacked on top of that one, but this one adds tests and fixes extra edge cases)

### CODE REVIEW

File: lib/clients/public.js [hunk 1]
```diff
@@ -58,17 +58,28 @@ class PublicClient {
...
-        return;
-      }
+
       if (err) {
         err.response = response;
-        reject(err);
-      }
-      if (data === null) {
+      } else if (response.statusCode > 299) {
+        err = new Error(`HTTP ${response.statusCode} Error`);
```
Comment (fb55) [inline lib/clients/public.js]: Use the `message` from the response here?

Comment (rmm5t) [reply lib/clients/public.js]: Do you mean the `statusMessage` from the `response`, or the `message` from the `body`, or both? I'm inclined to try both.

Comment (fb55) [reply lib/clients/public.js]: `message` from the body. `statusMessage` wouldn't be too bad either 🤷‍♂️

**Comment by rmm5t** [comment]
@fb55 I added the body's error message to the Error object's message. I skipped the `statusMessage`, because it's not all that beneficial anyway and [nock doesn't yet support mocking it at the moment](https://github.com/node-nock/nock/projects/2).

**Comment by fb55** [comment]
Amazing, thanks a lot!
